WinUIGallery/Helper/NavigationHelper.cs

```diff
@@ -215,14 +215,71 @@ private bool TryGoBack()
             return navigated;
         }
 
+        private bool TryGoForward()
+        {
+            bool navigated = false;
+            if (this.Frame.CanGoForward)
+            {
+                this.Frame.GoForward();
+                navigated = true;
+            }
+            return navigated;
+        }
+
         private void UpdateBackButton()
         {
-            if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 6))
+            this.CurrentNavView.IsBackEnabled = this.Frame.CanGoBack ? true : false;
+        }
+
+        /// <summary>
+        /// Invoked on every keystroke, including system keys such as Alt key combinations.
+        /// Used to detect keyboard navigation between pages even when the page itself
+        /// doesn't have focus.
+        /// </summary>
+        /// <param name="sender">Instance that triggered the event.</param>
+        /// <param name="e">Event data describing the conditions that led to the event.</param>
+
+        private void CurrentNavView_KeyDown(object sender, Microsoft.UI.Xaml.Input.KeyRoutedEventArgs e)
```

Comment:
I believe these keyboard events should be handled `OnKeyUp` so there won't be a need to handle `bool alreadyprocessedKeyDown` and `e.Handled`

Reply:
I also contemplated doing that but I was trying to implement the behavior as closely as possible to the WinUI 2 Gallery which does that on key down. Edge for example also navigates back on key down. What are your thoughts on changing the behavior?

Reply:
I see, in this case we can keep it as is.